homeassistant/components/hangouts/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""
+The hangouts bot component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hangouts/
+"""
+import logging
+
+from homeassistant import config_entries
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers import dispatcher
+
+from .config_flow import configured_hangouts
+from .const import (
+    CONF_BOT, CONF_COMMANDS, CONF_REFRESH_TOKEN, DOMAIN,
+    EVENT_HANGOUTS_CONNECTED, EVENT_HANGOUTS_CONVERSATIONS_CHANGED,
+    MESSAGE_SCHEMA, SERVICE_SEND_MESSAGE,
+    SERVICE_UPDATE_USERS_AND_CONVERSATIONS)
+
+REQUIREMENTS = ['hangups==0.4.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the Hangouts bot component."""
+    config = config.get(DOMAIN, [])
+    hass.data[DOMAIN] = {CONF_COMMANDS: config[CONF_COMMANDS]}
+
+    if configured_hangouts(hass) is None:
+        hass.async_add_job(hass.config_entries.flow.async_init(
+            DOMAIN, context={'source': config_entries.SOURCE_IMPORT}
+        ))
+
+    return True
+
+
+async def async_setup_entry(hass, config):
+    """Set up a config entry."""
+    from hangups.auth import GoogleAuthError
+
+    try:
+        from .hangouts_bot import HangoutsBot
+
+        bot = HangoutsBot(
+            hass,
+            config.data.get(CONF_REFRESH_TOKEN),
+            hass.data[DOMAIN][CONF_COMMANDS])
+        hass.data[DOMAIN][CONF_BOT] = bot
+    except GoogleAuthError as exception:
+        _LOGGER.error("Hangouts failed to log in: %s", str(exception))
+        return False
+
+    hass.bus.async_listen(EVENT_HANGOUTS_CONNECTED,
```

Comment:
This needs to use the dispatcher too

Reply:
are you sure? i fire the event with `hass.bus.async_fire`, so user can react if hangouts is connected (and also if its disconnected)